client/components/data/query-posts/index.jsx

```diff
@@ -14,61 +14,44 @@ import { isRequestingPostsForQuery, isRequestingSitePost } from 'calypso/state/p
  */
 const log = debug( 'calypso:query-posts' );
 
-class QueryPosts extends Component {
-	// @TODO: Please update https://github.com/Automattic/wp-calypso/issues/58453 if you are refactoring away from UNSAFE_* lifecycle methods!
-	UNSAFE_componentWillMount() {
-		this.request( this.props );
+const useMemoizedQuery = ( query ) => {
```

Comment:
Should we extract this and make it more generically reusable? I think it could be useful in other components as well.

Also, perhaps it could be useful if it supports multiple arguments, too (although multiple arguments could be passed as an array right now too)?

Nothing that has to be done in this PR though.